### PR TITLE
While replicating the transaction log, HazelcastInstanceNotActive exception is logged istead of throwing to the user

### LINKED ISSUE #3796: TransactionQueueTest.testIssue859And863
```
java.lang.AssertionError: expected:<2000> but was:<1999>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at com.hazelcast.queue.TransactionQueueTest.testIssue859And863(TransactionQueueTest.java:265)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/com.hazelcast$hazelcast/1803/testReport/junit/com.hazelcast.queue/TransactionQueueTest/testIssue859And863/

### PR DESCRIPTION
**Pull request by gurbuzali**
fixes #3796

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java [hunk 1]
```diff
@@ -466,6 +458,29 @@ public TransactionType getTransactionType() {
...
 
+    ExceptionHandler createReplicationTxExceptionHandler() {
+        return new ExceptionHandler() {
+            @Override
+            public void handleException(Throwable throwable) {
+                if (throwable instanceof TimeoutException) {
+                    throw new TransactionTimedOutException(throwable);
+                }
+                if (throwable instanceof MemberLeftException) {
+                    nodeEngine.getLogger(Transaction.class).warning("Member left while replicating tx begin: " + throwable);
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java]: minor issue: there's already a `logger` field in class.

**Comment by mdogan** [comment]
👍

File: hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java [hunk 2]
```diff
@@ -384,22 +391,7 @@ private void forceCreateBackupLogs() {
...
-                    e = e.getCause() != null ? e.getCause() : e;
-                }
-                if (e instanceof TargetNotMemberException) {
-                    nodeEngine.getLogger(Transaction.class).warning("Member left while replicating tx begin: " + e);
-                } else {
-                    RETHROW_TRANSACTION_EXCEPTION.handleException(e);
-                }
-            }
-        }
+        waitWithDeadline(futures, timeoutMillis, MILLISECONDS, replicationTxExceptionHandler);
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java]: there is a change in timeout - it user to be per-future, now the timeout is global for all futures - is this change intentional?

Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java]: yes, it is intentional. user should not wait more than configured transaction timeout.

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**